packages/integrations/cloudflare/src/server.directory.ts

```diff
@@ -38,17 +29,35 @@ export function createExports(manifest: SSRManifest) {
 				Symbol.for('astro.clientAddress'),
 				request.headers.get('cf-connecting-ip')
 			);
+
+			// @deprecated: getRuntime() can be removed, use `Astro.locals.runtime` instead
 			Reflect.set(request, Symbol.for('runtime'), {
-				...runtimeEnv,
+				...context,
 				waitUntil: (promise: Promise<any>) => {
-					runtimeEnv.waitUntil(promise);
+					context.waitUntil(promise);
 				},
 				name: 'cloudflare',
 				next,
 				caches,
 				cf: request.cf,
 			});
-			let response = await app.render(request, routeData);
+
+			let response = await app.render(request, routeData, {
+				env: env,
+				cf: request.cf,
+				runtime: {
+					// request: Request; // not needed because we have Astro.request, thus they are minor differences
+					// functionPath: string; // not needed
+					waitUntil: (promise: Promise<any>) => { context.waitUntil(promise); },
+					// passThroughOnException: () => void; // probably not needed ??
+					// next: (input?: Request | string, init?: RequestInit) => Promise<Response>; // probably not needed ??
+					env: context.env,
+					params: context.params, // Isn't that the same as Astro.props ??
+					// data: Data; // Should we include it ??: https://community.cloudflare.com/t/what-is-context-data-in-pages-functions/476559/7
```

Comment:
What is your opinion, should we include it for users who want to use it, I can't see any use-case.

Reply:
We can include them **if** users need it. For new, we can hide them and add them as we go if users request them.